smartsim/_core/generation/operations.py

```diff
@@ -0,0 +1,298 @@
+                "-m",
+                entry_point_path,
+                configure_cmd,
+                str(self.src),
+                final_dest,
+                self.tag,
+                self.file_parameters,
+            ]
+        )
+
+
+T = t.TypeVar("T", bound=GenerationProtocol)
+
+
+@dataclass
+class FileSysOperationSet:
+    """Dataclass to represent a set of file system operation objects"""
+
+    # TODO disallow modification - dunder function (post ticket to reevaluate API objects)
+    operations: t.List[GenerationProtocol] = field(default_factory=list)
+    """Set of file system objects that match the GenerationProtocol"""
+
+    def add_copy(
+        self, src: pathlib.Path, dest: t.Optional[pathlib.Path] = None
+    ) -> None:
+        """Add a copy operation to the operations list
+
+        :param src: Path to source
+        :param dest: Path to destination
+        """
+        self.operations.append(CopyOperation(src, dest))
+
+    def add_symlink(
+        self, src: pathlib.Path, dest: t.Optional[pathlib.Path] = None
+    ) -> None:
+        """Add a symlink operation to the operations list
+
+        :param src: Path to source
+        :param dest: Path to destination
+        """
+        self.operations.append(SymlinkOperation(src, dest))
+
+    def add_configuration(
+        self,
+        src: pathlib.Path,
+        file_parameters: t.Mapping[str, str],
+        dest: t.Optional[pathlib.Path] = None,
+        tag: t.Optional[str] = None,
+    ) -> None:
+        """Add a configure operation to the operations list
+
+        :param src: Path to source
+        :param file_parameters: File parameters to find and replace
+        :param dest: Path to destination
+        :param tag: Tag to use for find and replacement
+        """
+        self.operations.append(ConfigureOperation(src, file_parameters, dest, tag))
+
+    @property
+    def copy_operations(self) -> t.List[CopyOperation]:
+        """Property to get the list of copy files.
+
+        :return: List of CopyOperation objects
+        """
+        return self._filter(CopyOperation)
+
+    @property
+    def symlink_operations(self) -> t.List[SymlinkOperation]:
+        """Property to get the list of symlink files.
+
+        :return: List of SymlinkOperation objects
+        """
+        return self._filter(SymlinkOperation)
+
+    @property
+    def configure_operations(self) -> t.List[ConfigureOperation]:
+        """Property to get the list of configure files.
+
+        :return: List of ConfigureOperation objects
+        """
+        return self._filter(ConfigureOperation)
+
+    def _filter(self, type: t.Type[T]) -> t.List[T]:
+        """Filters the operations list to include only instances of the
+        specified type.
+
+        :param type: The type of operations to filter
+        :return: A list of operations that are instances of the specified type
+        """
+        return [x for x in self.operations if isinstance(x, type)]
```

Comment:
Just a heads up, pylint should complain about us "redefining builtin symbol `type`" if/when we turn that back on. Consider adding a trailing underscore or something to the symbol to avoid.
```suggestion
    def _filter(self, type_: t.Type[T]) -> t.List[T]:
        """Filters the operations list to include only instances of the
        specified type.

        :param type_: The type of operations to filter
        :return: A list of operations that are instances of the specified type
        """
        return [x for x in self.operations if isinstance(x, type_)]
```

Reply:
ah nice catch